File: GenericStack.cpp
#include <iostream>
#include <stack>
#include <string>
#include <stdexcept>

using namespace std; 

template<typename T>
class GenericStack {
private:
    stack<T> stack; 
public:
    // Constructor
    GenericStack() {}

    // Method to push elements onto the stack
    void push(T t) {
        stack.push(t);
    }

    // Method to pop elements from the stack
    T pop() {
        if (stack.empty()) {
            throw runtime_error("Stack is empty"); /
        }
        T top = stack.top();
        stack.pop();
        return top;
    }
};

int main() {
    // Create an instance of GenericStack
    GenericStack<int> stack; 
  
    stack.push(2);
    stack.push(2);
    stack.push(3);

    // Pop and print elements from the stack
    cout << stack.pop() << endl; // This will print 3
    cout << stack.pop() << endl; // This will print 2
    cout << stack.pop() << endl; // This will print 2

    return 0;
}

File: Generic-JAVA/GenericSTACK.java
import java.util.Stack;

public class GenericSTACK<T> {
    Stack<T> stack = new Stack<>();

    // Constructor
    public GenericSTACK() {}

    // Method to push elements onto the stack
    public void push(T t) {
        stack.push(t);
    }

    // Method to pop elements from the stack
    public T pop() {
        return stack.pop();
    }

    public static void main(String[] args) {
        // Create an instance of GenericSTACK
        GenericSTACK<Object> stack = new GenericSTACK<>();

        // Push elements onto the stack
        stack.push(2);
        stack.push(2.2);
        stack.push("String");

        // Pop and print elements from the stack
        System.out.println(stack.pop());//Thes will print String...
        System.out.println(stack.pop()); // This will print 2.2..
        System.out.println(stack.pop()); // This will print 2.
    }
}
